### PR TITLE
selfHostedEtcd: enable TLS

### PR DESCRIPTION
**Pull request by hongchaodeng**
Last step of https://github.com/kubernetes-incubator/bootkube/issues/582

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Current PR shows how TLS works.
It assumes we will take and only take secure etcd endpoint. I'm open to this since I don't have much experience at deploying.

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by xiang90** [comment]
LGTM.

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

File: cmd/bootkube/render.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ const (
 	bootEtcdOffset               = 20
 	defaultServiceBaseIP         = "10.3.0.0"
 	defaultEtcdServers           = "https://127.0.0.1:2379"
-	defaultSelfHostedEtcdServers = "http://127.0.0.1:2379"
+	defaultSelfHostedEtcdServers = "https://127.0.0.1:2379"
```
Comment (diegs) [inline cmd/bootkube/render.go]: should just remove this and use `defaultEtcdServers` now that they are the same.

**Comment by xiang90** [comment]
@diegs 

good point.

@hongchaodeng 

can you give recovery a try?

**Comment by hongchaodeng** [comment]
Sure. (got a rare full green..)

**Comment by hongchaodeng** [comment]
coreosbot run e2e

File: pkg/util/etcdutil/util.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func WaitClusterReady(endpoint string, etcdTLS *tls.Config) error {
 	return err
 }
 
-func makeTLSConfig(assetDir string) (*tls.Config, error) {
+func MakeTLSConfig(assetDir string) (*tls.Config, error) {
```
Comment (xiang90) [inline pkg/util/etcdutil/util.go]: NewTLSConfig. That follows go style better.

**Comment by hongchaodeng** [comment]
For recovery,  I think it would make more sense to _not_ use TLS to recover assets.
Moreover, it should also recover the etcd TLS assets.
I seemed to encounter some problem. Gonna fix and push it here on any update.

**Comment by hongchaodeng** [comment]
I have tested TLS works for recovery as well.
Will need https://github.com/kubernetes-incubator/bootkube/pull/603 to merge first

**Comment by xiang90** [comment]
The second commit needs a better commit message. Other than that, LGTM.

**Comment by hongchaodeng** [comment]
I would need to squash it.
Let me rebase after https://github.com/kubernetes-incubator/bootkube/pull/603 is merged

**Pull request merged**